bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/DeviceWithMeasureHandler.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class DeviceWithMeasureHandler extends DeviceHandler {
+    private final Logger logger = LoggerFactory.getLogger(DeviceWithMeasureHandler.class);
+    private final Optional<WeatherApi> weatherApi;
+    private final Map<String, State> measures = new HashMap<>();
+    private final Set<Channel> measureChannels = new HashSet<>();
+
+    public DeviceWithMeasureHandler(Bridge bridge, List<AbstractChannelHelper> channelHelpers, ApiBridge apiBridge,
+            NetatmoDescriptionProvider descriptionProvider) {
+        super(bridge, channelHelpers, apiBridge, descriptionProvider);
+
+        weatherApi = apiBridge.getWeatherApi();
+        channelHelpers.stream().filter(c -> c instanceof MeasuresChannelHelper).findFirst()
+                .map(MeasuresChannelHelper.class::cast).ifPresent(helper -> helper.setMeasures(measures));
+    }
+
+    @Override
+    public void setNewData(NAObject newData) {
+        collectMeasures();
+        getBridgeHandler().ifPresentOrElse(handler -> ((DeviceWithMeasureHandler) handler)
+                .callGetMeasurements(config.id, measureChannels, measures),
+                () -> callGetMeasurements(null, measureChannels, measures));
+        super.setNewData(newData);
+    }
+
+    private void callGetMeasurements(@Nullable String moduleId, Set<Channel> measureChannels,
+            Map<String, State> localMeasures) {
+        localMeasures.clear();
+        weatherApi.ifPresent(api -> {
+            measureChannels.forEach(channel -> {
+                Configuration measureDef = channel.getConfiguration();
+                String period = (String) measureDef.get("period");
+                String limit = (String) measureDef.get("limit");
+                ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+                if (channelTypeUID != null) {
+                    String channelId = channelTypeUID.getId();
+                    String[] elements = channelId.split("-");
+                    MeasureClass.asSet.stream().filter(mt -> mt.tagName.equals(elements[0])).findFirst()
+                            .ifPresent(mt -> {
+                                try {
+                                    Object result = api.getMeasurements(config.id, moduleId, period, mt, limit);
+
+                                    State data = result instanceof ZonedDateTime
+                                            ? toDateTimeType((ZonedDateTime) result)
+                                            : result instanceof Double ? toQuantityType((Double) result, mt)
+                                                    : UnDefType.UNDEF;
+
+                                    localMeasures.put(channel.getUID().getIdWithoutGroup(), data);
+                                } catch (NetatmoException e) {
+                                    logger.info("Error getting measurement {} for channel {}",
```

Comment:
Message says error, should it be logged as error?

Reply:
Done.